lib/sidebar_builder.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: false
+
+module SwedbankPay
+  # A nice sidebar
+  class SidebarBuilder
+    def initialize(filename, pages)
+      @filename = filename
+      @pages = pages
+    end
+
+    def build
+      sidebar_markup = ''
+
+      @pages.select { |path, _| path.split('/').length <= 2 }.each do |path, page|
+        next if page[:title].nil?
+        next if page[:hide_from_sidebar]
+
+        sidebar_markup << generate_main_markup(path, page)
+      end
+
+      sidebar_markup
+    end
+
+    private
+
+    def generate_main_markup(path, page)
+      title = page[:title].split('–').first
+      child_pages = @pages.select { |child_path, _| child_path.include?(path) && child_path != path && path != '/' }
```

Comment:
Codacy found an issue: [Line is too long. [116/80]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)